### PR TITLE
Added master_timeout parameter to CAT segments

### LINKED ISSUE #486: [BUG] Request parameter 'master_timeout' is missing in the page of 'CAT Segments' API
**What is the bug?**
The request parameter (or URL parameter) `master_timeout` is not shown in the page of `CAT Segments` API
https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-segments/

'master_timeout' is very a common parameter for OpenSearch REST APIs, and the meaning of this parameter is the same as in other APIs, such as https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-allocation/#url-parameters.

It's actually supported for `CAT Segments` API since OpenSearch version 1.0.0, and the source code is here: https://github.com/opensearch-project/OpenSearch/blob/1.0.0/server/src/main/java/org/opensearch/rest/action/cat/RestSegmentsAction.java#L84

**How can one reproduce the bug?**
n/a

**What is the expected behavior?**
'master_timeout' is added to the `URL Parameter` section of the page.

**What is your host/environment?**
n/a

**Do you have any screenshots?**
none.

**Do you have any additional context?**
none.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #486 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by hdhalter — APPROVED** [review]
Looks good!

**Comment by keithhc2** [comment]
Should we have two versions of this? One for 1.3, which still uses `master_timeout`, and then another version for 2.0, which would be `cluster_manager_timeout`.

**Comment by kolchfa-aws** [comment]
So, it's the same parameter but it's named differently in 2.0? If so, I can create two different versions.

**Comment by keithhc2** [comment]
> So, it's the same parameter but it's named differently in 2.0? If so, I can create two different versions.

Yeah, I believe that's a breaking change introduced in 2.0. You can look at #513 to see the exact syntax for what `cluster_manager` should be.

File: _opensearch/rest-api/cat/cat-segments.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ In addition to the [common URL parameters]({{site.url}}{{site.baseurl}}/opensear
 Parameter | Type | Description
 :--- | :--- | :---
 bytes | Byte size | Specify the units for byte size. For example, `7kb` or `6gb`. For more information, see [Supported units]({{site.url}}{{site.baseurl}}/opensearch/units/)..
+master_timeout | Time | The amount of time to wait for a connection to the master node. Default is 30 seconds.
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _opensearch/rest-api/cat/cat-segments.md:49]: And as Keith described, let's make a separate PR for 1.3

[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _opensearch/rest-api/cat/cat-segments.md:49]: OK thank you. Will do.

File: _opensearch/rest-api/cat/cat-segments.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ In addition to the [common URL parameters]({{site.url}}{{site.baseurl}}/opensear
 Parameter | Type | Description
 :--- | :--- | :---
 bytes | Byte size | Specify the units for byte size. For example, `7kb` or `6gb`. For more information, see [Supported units]({{site.url}}{{site.baseurl}}/opensearch/units/)..
+cluster_manager_timeout | Time | The amount of time to wait for a connection to the master node. Default is 30 seconds.
```
Comment (keithhc2) [inline _opensearch/rest-api/cat/cat-segments.md:49]: "cluster_manager" node?

**Review by keithhc2 — CHANGES_REQUESTED** [review]

File: _opensearch/rest-api/cat/cat-segments.md (hunk 2 above)
Comment (kolchfa-aws) [reply _opensearch/rest-api/cat/cat-segments.md:49]: In the description, should it be with or without underscore? 

The amount of time to wait for a connection to the cluster_manager node           vs
The amount of time to wait for a connection to the cluster manager node

Comment (kolchfa-aws) [reply _opensearch/rest-api/cat/cat-segments.md:49]: In #513 it says "master" is replaced with "cluster manager." So, I will change to "cluster manager" without the underscore unless there are any objections. Thank you!

**Review by keithhc2 — APPROVED** [review]
LGTM!

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**